### DISCUSSION TITLE
Cargo 16 - remake of Cargo map

### DISCUSSION THREAD

**Discussion by cloudziq** [redeclipse/base#1191] [Maps and Content Showcase]
Hello, since we have new 'forums', I present here my remake of Cargo map. (It's 'Kargo' for now - wip filename, so you can paste it safely into 'maps' folder without overwriting original map)

It's all WIP, most of the geometry is changed/moved, of course lighting is redone from scratch, texture work is not complete, but goal was to remove that old 'brownish' look. Map have bigger opening to allow more sun rays to come in, I replaced lava with acid, added some models, map have huge amount of lights (I'll write all stats later) but only these major ones are casting shadows, most of them are static ones. Major problem that I have encountered is that skylight is dependant of sun position on sky, so I moved sun almost to very top (please bring back old skylight ;)
Also earlier the red flag was in middle of corridor, while blue flag have its own room, so I added another room for red team. 
As I stated, layout is 95% completed so progress should be faster from now, mostly by optimising and texture work and making nicer blendmaps. Night version is half as complete and it is in bad state for now :\

In attachment you can find two folders: 'dziq_wip' and 'snipergoth', you need to put them in RE home folder or REinstall/data (preferably first option)

here are a few screens, better download and see for yourself:
<a href="https://imgur.com/mJW23eB"><img src="https://i.imgur.com/mJW23eBl.jpg" title="source: imgur.com" /></a>
<a href="https://imgur.com/kbQUEOw"><img src="https://i.imgur.com/kbQUEOwl.jpg" title="source: imgur.com" /></a>
<a href="https://imgur.com/P4lriUI"><img src="https://i.imgur.com/P4lriUIl.jpg" title="source: imgur.com" /></a>
<a href="https://imgur.com/H9jUsNN"><img src="https://i.imgur.com/H9jUsNNl.jpg" title="source: imgur.com" /></a>

What do you think about lighting, texturing, etc?
Any suggestions are welcome while this is all in WIP state, cheers :)

old version: [KARGO.zip](https://github.com/red-eclipse/world/files/1590329/KARGO.zip)
second revision (outdated): [Cargo16.zip](https://github.com/red-eclipse/world/files/2064556/Cargo16.zip)

check further posts to download latest revision

**Comment by Calinou** [comment]
This looks nice, but some areas seem a bit dark to me, perhaps make the lights brighter and/or larger in these areas. It's important to make sure the game is playable on all kinds of monitors, not just ones with poor contrast rates 😃

**Comment by qreeves** [comment]
> This looks nice, but some areas seem a bit dark to me, perhaps make the lights brighter and/or larger in these areas. It's important to make sure the game is playable on all kinds of monitors, not just ones with poor contrast rates

I'd have to agree with this.

**Comment by ghost** [comment]
From the screenshots, this looks really nice. Although I have noticed a few walls have been changed in such a way that would limit parkour as well as the addition of some objects (such as crates) and the creation of new routes, some play testing would be needed to make sure these changes don't negatively impact the flow of the map or cause unnecessary restriction of movement.

**Comment by cloudziq** [comment]
Maybe it's more detailed, but overall there is more places to do parkour than in original Cargo, every place where parkour is possible in original, it is also possible here in the same spots.
As for crates - yeah, there a few of those but still plenty of place to navigate, also these helping a little to take a cover

**Comment by cloudziq** [comment]
Hey guys and ladies, again I need a few words from our community and I have a bad feeling about all this but I decided to write.. 
Since RE 1.9, I specifically 'allergic' about map performance. Old Cargo map was made for lightweight engine running on calculators, with low-poly design in mind. Nowadays after more and more toying with this map, adding stuff, remixing, I think its not good in overall performance around most of the map
After some inspecting, /testpvs (2-128) I noticed many problems with occlusion. please look below to see one of those

place from where I tested with /testpvs 2 (yes, even that low):
![20180401031715](https://user-images.githubusercontent.com/13541407/38168989-96ce3534-355d-11e8-8e04-9384620eb42b.jpg)

And look how many surfaces are still there on the other side (view from almost other side of the map - central area):
![20180401031734](https://user-images.githubusercontent.com/13541407/38168992-c734e862-355d-11e8-8b88-bb95190e4f7d.jpg)

Also the same result is with /dtoutline 0 (seems so)
Now I confused because I have normally no problem with this, I can rearrange layout a little to make sure that occlusion is efficient, but this map is different.. many people will yell/bite/kill me for doing so masive changes ;p    so for now, its all up to you people...?    

@DEVS - Question of the month:  does checking how occlusion works with /dtoutline 0 and /testpvs give us 'proper' results?

**Comment by qreeves** [comment]
Seems to be a common problem with maps that use extreme levels of detail, this is why I think we need to adopt a simpler and cleaner look with brighter colours, these play really well with global illumination and creates a better atmosphere.

**Comment by cloudziq** [comment]
I can go around this and remix layout to be only similar to Cargo (to drastically speed up a map), but for now I searching a way to don't do this, so I think the 'simple and cleaner' is the way to go with this map, otherwise It's going to be running bad. 

edit:
@molexted, we need yours measuring-compar'o machine to look at performance problems in this map, I know that wide open area is problem but I have slow machine, maybe everything is all right and I unnecessary panicking (?)

**Comment by no-lex** [comment]
On my r5 340 (effectively identical to an  A8 apu) I'm getting 35-45 fps at min settings 900p, and 55-60 fps at min settings + renderscale 75% 900p. Not great, worse than ctf13 by 25-35%, but considering the lag-prone center with poor occlusion and water, it's not too bad.

Addressing base/818 should help lower these performance issues when it gets implemented.

**Comment by cloudziq** [comment]
I managed to remove problems in that area :) Now, from the same spot as before, much less faces is rendered (but map layout again is changed, some corridors are 8 meters longer than before, but in exchange map is much better occluded and running faster now), it is better optimised for Tesseract rendering pipeline -

again shot from central area after 'fixes':
![20180404005725](https://user-images.githubusercontent.com/13541407/38284940-268442f6-37be-11e8-9d33-2e0b29c6d8da.jpg)

That acid pool piece is interesting, it's rendered almost always during gameplay. hmmm

**Comment by cloudziq** [comment]
Latest version, sorry for no screenshots for now but barely anyone looking here, it is more for people who want to see all assets when coop.  Unpack, move and replace old files if any, :)
From now, map fully supports a /mapeffects feature
[Cargo16.zip](https://github.com/red-eclipse/world/files/2064556/Cargo16.zip)

**Comment by qreeves** [comment]
Now in master for testing purposes as of red-eclipse/base@8a3ec68e75e1296f19aad884668b6719f2712d91

**Comment by cloudziq** [comment]
uh oh long time without a update.. So long that map was even removed from official rotation
(because of two missing textures.. Thanks to molex and his plan to keep his ugly maps only, in 2,0-the-worst-ever-release)
Anyway, I not going to update first post. Instead I put some screens here to compare.

Little info about 9th revision of Cargo16:
Last time I changed texturing, especially floors to have brighter texture, so these surfaces are a bit unfinished - still need some tweaks.
Most spotlights were removed, lighting in general is a bit worse than before, but still acceptable - and what is more important - map runs a bit faster.  /mapeffects 2 will remove some eye candies on map like reducing particles by half, some small lights etc, while /mapeffects 1 can help further with that by removing decals, not needed mapmodels (for gameplay) and turn off particles entirelly.


![20190813045029](https://user-images.githubusercontent.com/13541407/62912376-3130e280-bd88-11e9-8e36-5074cda5f68f.jpg)
![20190813045210](https://user-images.githubusercontent.com/13541407/62912391-3c840e00-bd88-11e9-8981-5ebd8c9ea7ac.jpg)
![20190813045313](https://user-images.githubusercontent.com/13541407/62912396-40b02b80-bd88-11e9-86eb-5ac8450ca688.jpg)

map files to download:  [kargo09.zip](https://github.com/red-eclipse/world/files/3495151/kargo09.zip)

**Comment by cloudziq** [comment]
latest version of (formerly) Kargo,  a Cargo 16 (tribute to Deck 16 from UT99)
latest fixes were specifically done about better performance and aesthetics. Waypoints are set so can be played offline.   If your system choke on this map, try /mapeffects 1
.
[cargo16.zip](https://github.com/redeclipse/world/files/4191339/cargo16.zip)
.
much thanks for: (in alphabetical order as always) dc for background work (outside areas), Puzzlecube (for good support and ideas), and Unixfreak for helping with texture work and nice details + good advice
Special thanks to:  Hidan (for more than two years of fun in new RE), Jopsy (for planting a palmsy ;)
